### PR TITLE
Simplify OBX2 string mapping

### PR DESCRIPTION
**Pull request by thetaurean**
This PR cleans up the OBX2 mapping to remove a condition that was added to satisfy CO pilot.

Test Steps:
1. Run translation tests (`gradlew testIntegration`)

## Changes
- Removed dependency check on OBX5 within OBX2 mapping.

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] ~~Added~~ Modified tests

## Linked Issues
- #10943

### CODE REVIEW

**Review by jimmyfagan — CHANGES_REQUESTED** [review]
The change looks good to me, however it breaks several integration tests. You'll need to determine whether to update those integration tests so they pass, or adjust the schemas/transforms to keep them working. My gut says we should update the expected output for the failing tests, the input for the failing tests probably came from the same sample messages as the CO ones that were removed, so the behavior here is likely not reflective of anything happening with SimpleReport.

**Comment by thetaurean** [comment]
> The change looks good to me, however it breaks several integration tests. You'll need to determine whether to update those integration tests so they pass, or adjust the schemas/transforms to keep them working. My gut says we should update the expected output for the failing tests, the input for the failing tests probably came from the same sample messages as the CO ones that were removed, so the behavior here is likely not reflective of anything happening with SimpleReport.

That's exactly the green light I was looking for 😀 thanks for the sanity check.
I'll get those tests fixed up and re-request.

**Comment by thetaurean** [comment]
Tests fixed!

**Review by arnejduranovic — APPROVED** [review]

File: prime-router/metadata/hl7_mapping/common/observation-result.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ elements:
     hl7Spec: [ '%{hl7OBXField}-1' ]
 
   - name: result-value-type-st
-    condition: '(%resource.value.exists() and %resource.value.is(string)) or (%resource.dataAbsentReason.coding.where(extension(%`rsext-code-index-name`).value = "identifier").exists() and %resource.dataAbsentReason.coding.where(extension(%`rsext-code-index-name`).value = "identifier").code = "unknown")'
+    condition: '%resource.value is String'
```
Comment (jimmyfagan) [inline prime-router/metadata/hl7_mapping/common/observation-result.yml:7]: After merging, due to the HAPI FHIR updates [here](https://github.com/CDCgov/prime-reportstream/pull/11406), this condition is no longer valid and smoke tests and integration tests are now failing. Kalish posted some tips [here](https://nava.slack.com/archives/C01HFHPNJEL/p1695841059693489) (not sure if the link is Nava-specific or if it'll work for you), but you should be able to just keep what's currently in `master` with the last part removed like before.

```suggestion
    condition: '%resource.value.exists() and %resource.value.is(string)'
```

Comment (thetaurean) [reply prime-router/metadata/hl7_mapping/common/observation-result.yml]: Thankyou for the suggested fix!

**Review by jimmyfagan — APPROVED** [review]
Latest commits including merges look good!

**Pull request merged**